journalbeat/reader/journal.go

```diff
@@ -303,31 +287,19 @@ func (r *Reader) toEvent(entry *sdjournal.JournalEntry) *beat.Event {
 	return &event
 }
 
-func isKept(key string) bool {
-	return key != ""
-}
-
-// stopOrWait waits for a journal event.
-func (r *Reader) stopOrWait() {
-	select {
-	case <-r.done:
-	case r.changes <- r.journal.Wait(100 * time.Millisecond):
-	}
-}
-
 func (r *Reader) wait() {
 	select {
 	case <-r.done:
 		return
-	case <-time.After(r.config.Backoff):
+	case <-time.After(r.backoff):
 	}
 
-	if r.config.Backoff < r.config.MaxBackoff {
-		r.config.Backoff = r.config.Backoff * time.Duration(r.config.BackoffFactor)
-		if r.config.Backoff > r.config.MaxBackoff {
-			r.config.Backoff = r.config.MaxBackoff
+	if r.backoff < r.config.MaxBackoff {
+		r.backoff = r.backoff * time.Duration(r.config.BackoffFactor)
+		if r.backoff > r.config.MaxBackoff {
+			r.backoff = r.config.MaxBackoff
 		}
-		r.logger.Debugf("Increasing backoff time to: %v factor: %v", r.config.Backoff, r.config.BackoffFactor)
+		r.logger.Debugf("Increasing backoff time to: %v factor: %v", r.backoff, r.config.BackoffFactor)
 	}
 }
 
```

Comment:
instead of using a custom backoff implementations we might want to use [common.Backoff](https://github.com/elastic/beats/blob/master/libbeat/common/backoff.go), which implements everything that you have in this code. I missed that in the original review.